jerry-debugger/jerry_client_main.py

```diff
@@ -1151,33 +1163,34 @@ def _process_incoming_text(self, buffer_type, data):
                 log_type = "%sout:%s " % (self.blue, self.nocolor)
 
                 message = self.current_out + message
-                lines = message.split("\n")
-                self.current_out = lines.pop()
+                lines = message.decode("utf8").split("\n")
+                self.current_out = lines.pop().encode("utf8")
 
                 return "".join(["%s%s\n" % (log_type, line) for line in lines])
 
             if subtype == JERRY_DEBUGGER_OUTPUT_DEBUG:
                 log_type = "%slog:%s " % (self.yellow, self.nocolor)
 
                 message = self.current_log + message
-                lines = message.split("\n")
-                self.current_log = lines.pop()
+                lines = message.decode("utf8").split("\n")
+                self.current_log = lines.pop().encode("utf8")
 
                 return "".join(["%s%s\n" % (log_type, line) for line in lines])
 
             if not message.endswith("\n"):
                 message += "\n"
 
             if subtype == JERRY_DEBUGGER_OUTPUT_WARNING:
-                return "%swarning: %s%s" % (self.yellow, self.nocolor, message)
+                return "%swarning: %s%s" % (self.yellow, self.nocolor, message.decode("utf8"))
             elif subtype == JERRY_DEBUGGER_OUTPUT_ERROR:
-                return "%serr: %s%s" % (self.red, self.nocolor, message)
+                return "%serr: %s%s" % (self.red, self.nocolor, message.decode("utf8"))
             elif subtype == JERRY_DEBUGGER_OUTPUT_TRACE:
-                return "%strace: %s%s" % (self.blue, self.nocolor, message)
+                return "%strace: %s%s" % (self.blue, self.nocolor, message.decode("utf8"))
 
         # Subtypes of eval
         self.prompt = True
 
+        message = message.decode("utf8")
```

Comment:
This decode is only required if the `buffer_type == JERRY_DEBUGGER_OUTPUT_RESULT_END` check above is not true.

Reply:
Thanks, I've applied your change from https://gist.github.com/galpeter/a6ac851b6ab113aaad62d60001c5df23.